hilti/toolchain/src/compiler/validator.cc

```diff
@@ -725,6 +744,8 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
         std::set<ID> seen;
 
         for ( const auto& f : n->fields() ) {
+            if ( auto rc = checkIfProperlyNamed(f->type()); ! rc )
+                error(fmt("type cannot be used as field type (because %s)", rc.error()), f->type());
```

Comment:
I believe it should work to move this into its own visitor instead. Using the function from above:

```
void operator()(declaration::Field* n) final { checkDeclarationType(n, n->type()); }
```